nomad/job_endpoint_hook_connect_test.go

```diff
@@ -0,0 +1,81 @@
+package nomad
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/hashicorp/nomad/nomad/structs"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_isSidecarForService(t *testing.T) {
+	cases := []struct {
+		t *structs.Task //task
+		s string        // service
+		r bool          // result
+	}{
+		{
+			&structs.Task{},
+			"api",
+			false,
+		},
+		{
+			&structs.Task{
+				Kind: "connect-proxy:api",
+			},
+			"api",
+			true,
+		},
+		{
+			&structs.Task{
+				Kind: "connect-proxy:api",
+			},
+			"db",
+			false,
+		},
+		{
+			&structs.Task{
+				Kind: "api",
+			},
+			"api",
+			false,
+		},
+	}
+
+	for _, c := range cases {
+		require.Equal(t, c.r, isSidecarForService(c.t, c.s))
+	}
+}
+
+func Test_groupConnectHook(t *testing.T) {
+	tgIn := &structs.TaskGroup{
+		Networks: structs.Networks{
+			{
+				Mode: "bridge",
+			},
+		},
+		Services: []*structs.Service{
+			{
+				Name:      "backend",
+				PortLabel: "8080",
+				Connect: &structs.ConsulConnect{
+					SidecarService: &structs.ConsulSidecarService{},
+				},
+			},
+		},
+	}
+
+	tgOut := tgIn.Copy()
+	tgOut.Tasks = []*structs.Task{
+		newConnectTask(tgOut.Services[0]),
+	}
+	tgOut.Networks[0].DynamicPorts = []structs.Port{
+		{
+			Label: fmt.Sprintf("%s-%s", structs.ConnectProxyPrefix, "backend"),
+			To:    -1,
+		},
+	}
+
+	groupConnectHook(tgIn)
+	require.Exactly(t, tgOut, tgIn)
+}
```

Comment:
Add tests at the Mutate() and Validate() method levels to get a little more coverage.

Reply:
I added a new test on the register endpoint for a connect job